srtcore/core.cpp

```diff
@@ -10029,8 +10029,8 @@ int srt::CUDT::processData(CUnit* in_unit)
                 continue;
             }
 
-            const int avail_bufsize = getAvailRcvBufferSizeNoLock();
-            if (offset >= avail_bufsize)
+            const size_t avail_bufsize = getAvailRcvBufferSizeNoLock();
+            if (size_t(offset) >= avail_bufsize)
```

Comment:
Although there is a check for the `offset` sign above, I think it would be safer to add a debug assertion right above this casting.
Also, there is nothing bad in having `avail_bufsize` as `int`, because you would not expect RCV buffer size to be larger than 2147483647 packets (1316 or more bytes each).
```suggestion
            SRT_ASSERT(offset >= 0);
            const size_t avail_bufsize = getAvailRcvBufferSizeNoLock();
            if (size_t(offset) >= avail_bufsize)
```